lib/client-assets.php

```diff
@@ -770,10 +780,52 @@ function gutenberg_enqueue_registered_block_scripts_and_styles() {
 			wp_enqueue_script( $block_type->editor_script );
 		}
 	}
+
+	if ( $enqueue_only_required_styles ) {
+		add_action( 'the_content', 'enqueue_required_frontend_block_styles', 10 ); 
```

Comment:
Yes. Thanks for pointing this out. I'm sorry this completely skipped my mind.

Now, `gutenberg_process_block_comments` runs with `PHP_INT_MAX - 1` priority and `gutenberg_enqueue_frontend_styles` runs with `PHP_INT_MAX` priority since front-end styles can be enqueued once block types are parsed and saved in the registry.